src/internal/operations/manifests.go

```diff
@@ -15,11 +15,44 @@ import (
 	"github.com/alcionai/corso/src/pkg/path"
 )
 
-// calls kopia to retrieve prior backup manifests, metadata collections to supply backup heuristics.
-// TODO(ashmrtn): Make this a helper function that always returns as much as
-// possible and call in another function that drops metadata and/or
-// kopia-assisted incremental bases based on flag values.
 func produceManifestsAndMetadata(
+	ctx context.Context,
+	bf inject.BaseFinder,
+	rp inject.RestoreProducer,
+	reasons, fallbackReasons []kopia.Reasoner,
+	tenantID string,
+	getMetadata, getAssistBases bool,
+) (kopia.BackupBases, []data.RestoreCollection, bool, error) {
+	bb, meta, useMergeBases, err := getManifestsAndMetadata(
+		ctx,
+		bf,
+		rp,
+		reasons,
+		fallbackReasons,
+		tenantID,
+		getMetadata)
+	if err != nil {
+		return nil, nil, false, clues.Stack(err).OrNil()
```

Comment:
we already know the err isn't nil.
```suggestion
		return nil, nil, false, clues.Stack(err)
```